docs/memory_layout.md

```diff
@@ -0,0 +1,230 @@
+|----------------|----------------------------------------------------------------------------------------------------------------------------|
+| mat2x2\<int8>  | <kbd>[bb--][----][----][----]<br/>[bb--][----][----][----]                                                          </kbd> |
+| mat2x2\<fp16>  | <kbd>[bbbb][----][----][----]<br/>[bbbb][----][----][----]                                                          </kbd> |
+| mat2x2\<float> | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]                                                          </kbd> |
+| mat2x3\<float> | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]                                                          </kbd> |
+| mat2x4\<float> | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb][bbbb][bbbb]                                                          </kbd> |
+| mat3x2\<float> | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]                             </kbd> |
+| mat4x2\<float> | <kbd>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]<br/>[bbbb][bbbb][----][----]</kbd> |
+| mat4x3\<float> | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]</kbd> |
+
+### STD430
+| Name           | Bytes                                                                                                                      |
+|----------------|----------------------------------------------------------------------------------------------------------------------------|
+| mat2x2\<fp16>  | <kbd>[bbbb][bbbb]                                                                                                   </kbd> |
+| mat2x2\<float> | <kbd>[bbbb][bbbb][bbbb][bbbb]                                                                                       </kbd> |
+| mat2x3\<float> | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]                                                          </kbd> |
+| mat2x4\<float> | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb][bbbb][bbbb]                                                          </kbd> |
+| mat2x2\<int8>  | <kbd>[bbbb]                                                                                                         </kbd> |
+| mat3x2\<float> | <kbd>[bbbb][bbbb]<br/>[bbbb][bbbb]<br/>[bbbb][bbbb]                                                                 </kbd> |
+| mat3x3\<float> | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]                             </kbd> |
+| mat4x2\<float> | <kbd>[bbbb][bbbb][bbbb][bbbb]<br/>[bbbb][bbbb][bbbb][bbbb]                                                          </kbd> |
+| mat4x3\<float> | <kbd>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]<br/>[bbbb][bbbb][bbbb][----]</kbd> |
+
+
+## Matrix Array (All matrices are column matrices)
```

Comment:
Probably worth noting that the arraystride equals the base alignment of the matrix.  So there is no padding what is shown here and the next element in the array.

Reply:
done